src/partition/ReceivedBoundingBox.cpp

```diff
@@ -0,0 +1,200 @@
+  { 
+    //some code here
+  }
+
+  if (utils::MasterSlave::_masterMode)
+  { // Master
+    assertion(utils::MasterSlave::_rank==0);
+    assertion(utils::MasterSlave::_size>1);
+
+    // sending this participant communicator size to other master
+    _m2n->getMasterCommunication()->send(utils::MasterSlave::_size , 0);
+    utils::MasterSlave::_communication->broadcast(_remoteParComSize);
+    com::CommunicateBoundingBox(utils::MasterSlave::_communication).broadcastSendBoundingBoxMap(_globalBB);
+
+    // initializing _feedbackmap
+    for (int rank_slave=1; rank_slave < utils::MasterSlave::_size ; rank_slave++)
+    {
+      _feedbackMap[rank_slave].push_back(-1);        
+    }
+
+    // master produces its own feedback
+    for (auto &other_rank: _globalBB)
+    {
+      if (compareBoundingBox(_bb,other_rank.second))
+      {
+        _feedback.push_back(other_rank.first);
+      }          
+    }
+
+    // master first adds ist feedback to feedbackmap
+    _feedbackMap[0]=_feedback;
+
+    // master receives feedbacks from slaves and put them into the feedbackmap
+    for (int rank_slave=1; rank_slave < utils::MasterSlave::_size ; rank_slave++)
+    {
+      utils::MasterSlave::_communication->receive(_feedback, rank_slave);         
+      _feedbackMap[rank_slave]=_feedback;
+    }
+
+    // master sends feedbackmap to other master
+    com::CommunicateBoundingBox(_m2n->getMasterCommunication()).sendFeedbackMap(_feedbackMap,0);             
+  }
+  else if (utils::MasterSlave::_slaveMode)
+  {
+    utils::MasterSlave::_communication->broadcast(_remoteParComSize, 0);
+
+    // initializing the _globalBB
+    for (int remoteRank = 0; remoteRank < _remoteParComSize; remoteRank++ )
+    {
+      _globalBB[remoteRank]= _bb;
+    }
+    
+    // receive _globalBB from master
+    com::CommunicateBoundingBox(utils::MasterSlave::_communication).broadcastReceiveBoundingBoxMap(_globalBB);
+
+
+    for (auto &other_rank: _globalBB)
+    {
+      if (compareBoundingBox(_bb,other_rank.second)) {
+        _feedback.push_back(other_rank.first);            
+      }
+    }
+
+    // to prevent sending empty vector!
+    if(_feedback.size() == 0)
+    {
+      _feedback.push_back(-1);
+    }
+
+    //send feedback to master
+    utils::MasterSlave::_communication->send(_feedback, 0);
+    
+  }  
+}
+
+bool ReceivedBoundingBox::compareBoundingBox(mesh::Mesh::BoundingBox currentBB, mesh::Mesh::BoundingBox receivedBB)
+{
+  //int sizeofBB = currentBB.size();
+  bool intersect= true;
+
+  for (int i=0; i < _dimensions; i++) {
+
+    /*
+     * here two boundingboxs are compared to check there is an intersection or not! comparison is done for 
+     * all dimensions and in order to have intersection, at all dimensions there should be a common region!
+     */
+    if ((currentBB[i].first < receivedBB[i].first && currentBB[i].second < receivedBB[i].first) || (receivedBB[i].first < currentBB[i].first && receivedBB[i].second < currentBB[i].first) ) {
```

Comment:
Can you refactor this into another predicate?

Reply:
i put it on two lines, now it should be easier to read